src/app/pipes/filter/filter.pipe.ts

```diff
@@ -1,18 +1,37 @@
-import { Pipe, PipeTransform } from '@angular/core';
+import {
+  Pipe,
+  PipeTransform,
+} from '@angular/core';
 
 @Pipe({
   name: 'filter',
 })
 export class FilterPipe implements PipeTransform {
 
-  transform(items: any, filter: any): any {
+  /**
+   * -Si- la key présente dans filter est présente dans fullMatch
+   * -Alors- on vérifie que la valeur est exact
+   * -Sinon- on vérifie qu'elle contient cette valeur
+   *
+   * Exemple :
+   * filter:{type: "board", nom: "pawn"}:["type"]
+   * Donne les objets de type board (totoboard marche pas)
+   * qui contient pawn dans leur nom (whitepawn, blackpawn, anepawnpawn marche)
+   */
+  transform(items: any, filter: any, fullMatch = []): any {
     if (!filter)
       return items;
     const keys = Object.keys(filter);
+
     return items.filter((item) => {
-      const bol = keys.every((key) => item.hasOwnProperty(key) && item[key] === filter[key]);
+      const bol = keys.every((key) => {
```

Comment:
pourquoi ne pas mettre le return ici directement ?

Reply:
why not